Application/Sources/Settings/Service.swift

```diff
@@ -5,84 +5,91 @@
-    )
+    var id: Self {
+        self
+    }
 
-    static var samStage = Self(
-        id: "sam stage",
-        name: "SAM \(NSLocalizedString("Stage", comment: "Server setting name"))",
-        url: SRGIntegrationLayerStagingServiceURL().appendingPathComponent("sam")
-    )
+    var environment: String {
+        rawValue
+    }
 
-    static var samTest = Self(
-        id: "sam test",
-        name: "SAM \(NSLocalizedString("Test", comment: "Server setting name"))",
-        url: SRGIntegrationLayerTestServiceURL().appendingPathComponent("sam")
-    )
+    var name: String {
+        switch self {
+        case .production:
+            NSLocalizedString("Production", comment: "Server setting name")
+        case .stage:
+            NSLocalizedString("Stage", comment: "Server setting name")
+        case .test:
+            NSLocalizedString("Test", comment: "Server setting name")
+        case .mmf:
+            "Play MMF"
+        case .samProduction:
+            "SAM \(NSLocalizedString("Production", comment: "Server setting name"))"
+        case .samStage:
+            "SAM \(NSLocalizedString("Stage", comment: "Server setting name"))"
+        case .samTest:
+            "SAM \(NSLocalizedString("Test", comment: "Server setting name"))"
+        }
+    }
 
-    static var services: [Self] = [production, stage, test, mmf, samProduction, samStage, samTest]
+    var url: URL {
+        switch self {
+        case .production:
+            ApplicationConfiguration().dataProviderProductionServiceURL
+        case .stage:
+            ApplicationConfiguration().dataProviderStageServiceURL
+        case .test:
+            ApplicationConfiguration().dataProviderTestServiceURL
+        case .mmf:
+            Self.mmfUrl
+        case .samProduction:
+            ApplicationConfiguration().dataProviderProductionServiceURL.appendingPathComponent("sam")
+        case .samStage:
+            ApplicationConfiguration().dataProviderStageServiceURL.appendingPathComponent("sam")
+        case .samTest:
+            ApplicationConfiguration().dataProviderTestServiceURL.appendingPathComponent("sam")
+        }
+    }
 
-    static func service(forId id: String?) -> Self {
+    static func service(for environment: String?) -> Self {
         #if DEBUG || NIGHTLY || BETA
-            guard let id, let server = services.first(where: { $0.id == id }) else {
+            guard let environment, let service = Self(rawValue: environment) else {
                 return .production
             }
-            return server
+            return service
         #else
             return .production
         #endif
     }
 }
 
 @objc class ServiceObjC: NSObject {
-    @objc static func name(forServiceId serviceId: String) -> String {
-        Service.service(forId: serviceId).name
+    @objc static var environments = Service.allCases.map(\.environment)
+
+    @objc static func name(for environment: String) -> String {
```

Comment:
```suggestion
    @objc static func name(forEnvironment environment: String) -> String {
```

This better matches usual Objective-C conventions.

Reply:
Nice findings. Back to ObjC :)